app/src/org/commcare/activities/LoginActivityUIController.java

```diff
@@ -377,13 +405,13 @@ private void setStyleDefault() {
         passwordOrPin.setCompoundDrawablesWithIntrinsicBounds(getResources().getDrawable(R.drawable.icon_lock_neutral50), null, null, null);
         setupLoginButton();
         if (loginButton.isEnabled()) {
-            // don't hide error box when showing permission error
-            errorBox.setVisibility(View.GONE);
+            clearErrorMessage();
         }
     }
 
     protected void clearErrorMessage() {
-        errorBox.setVisibility(View.GONE);
+        errorContainer.setVisibility(View.GONE);
+        notificationButtonView.setVisibility(View.GONE);
```

Comment:
Shouldn't `errorContainer.setVisibility(View.GONE)` also takes care of `notificationButtonView` since it's a child of errorContainer (Not putting this for other use cases, but should get corrected for install and update as well if we do it here)

Reply:
For login specifically, the error container sometimes contains only an error message. When things pop back up, if we were previously displaying an error message and the button, the next time it may only contain an error message, so we need both to have their own logic.